src/hooks/useSingleProduct.ts

```diff
@@ -5,6 +5,16 @@ interface SingleProductApiData {
   body: Product;
 }
 
+class HttpError extends Error {
+  status: number;
+  statusText: string;
+  constructor(message: string, status: number, statusText: string) {
+    super(message);
+    this.status = status;
+    this.statusText = statusText;
+  }
+}
```

Comment:
Meant to mention this when we were on the call yesterday - it probably makes sense to define this class in `src/utilities` somewhere. Maybe a file `errors.ts` where we can define other custom errors we may need. This class would be useful all over the application, as opposed to just in this hook.